corehq/apps/app_manager/views/utils.py

```diff
@@ -415,6 +415,37 @@ def form_has_submissions(domain, app_id, xmlns):
     return FormES().domain(domain).app([app_id]).xmlns([xmlns]).count() != 0
 
 
+def get_new_multimedia_between_builds(domain, target_build_id, source_build_id, build_profile_id=None):
+    source_build = get_app_cached(domain, source_build_id)
+    target_build = get_app_cached(domain, target_build_id)
+    assert source_build.copy_of, _("Size calculation available only for builds")
+    assert target_build.copy_of, _("Size calculation available only for builds")
+    build_profile = source_build.build_profiles[build_profile_id] if build_profile_id else None
+    multimedia_map_for_source_build_by_id = {
+        media_item['multimedia_id']: media_item
+        for path, media_item in
+        source_build.multimedia_map_for_build(build_profile=build_profile).items()
+    }
+    multimedia_map_for_target_build = target_build.multimedia_map_for_build(build_profile=build_profile)
+    multimedia_map_for_target_build_by_id = {
+        media_item['multimedia_id']: media_item
+        for path, media_item in
+        multimedia_map_for_target_build.items()
+    }
```

Comment:
Could be reduced a bit by extracting a function:
```python
def _get_mm_map_by_id(build, profile):
    return {
        media_item['multimedia_id']: media_item
        for path, media_item in
        build.multimedia_map_for_build(build_profile=profile).items()
    }

multimedia_map_for_source_build_by_id = _get_mm_map_by_id(source_build, build_profile)
multimedia_map_for_target_build_by_id = _get_mm_map_by_id(target_build, build_profile)
```

Reply:
Could not use it exactly because i needed the target mm map later.
https://github.com/dimagi/commcare-hq/pull/25868/commits/704c36ba8e1c4264e32acf76601117634593cc1a